grails-app/controllers/org/pih/warehouse/core/DocumentController.groovy

```diff
@@ -207,10 +210,17 @@ class DocumentController {
                     contentType: file.contentType,
                     extension: FileUtil.getExtension(file.originalFilename),
                     documentNumber: command.documentNumber,
-                    documentType: DocumentType.get(typeId))
+                    documentType: documentType)
+
+            documentInstance.validate()
+
+            List<DocumentType> nonTemplateDocumentTypes = documentService.getNonTemplateDocumentTypes()
+            if (!nonTemplateDocumentTypes.contains(documentType)) {
```

Comment:
this could be handled in the `validator` of the `Document`, but since we are tight on time and the guys approved, I will merge it as is.